extensions/istio/model-v1alpha3/src/main/java/io/fabric8/istio/api/networking/v1alpha3/CorsPolicy.java

```diff
@@ -0,0 +1,199 @@
+/**
+ * Copyright (C) 2015 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.fabric8.istio.api.networking.v1alpha3;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import com.fasterxml.jackson.annotation.JsonAnyGetter;
+import com.fasterxml.jackson.annotation.JsonAnySetter;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+
+import io.fabric8.kubernetes.api.model.Container;
+import io.fabric8.kubernetes.api.model.ContainerPort;
+import io.fabric8.kubernetes.api.model.EnvVar;
+import io.fabric8.kubernetes.api.model.KubernetesResource;
+import io.fabric8.kubernetes.api.model.LabelSelector;
+import io.fabric8.kubernetes.api.model.ObjectMeta;
+import io.fabric8.kubernetes.api.model.ObjectReference;
+import io.fabric8.kubernetes.api.model.Volume;
+import io.fabric8.kubernetes.api.model.VolumeMount;
+import io.sundr.builder.annotations.Buildable;
+import io.sundr.builder.annotations.BuildableReference;
+import lombok.EqualsAndHashCode;
+import lombok.ToString;
+
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonPropertyOrder({
+  "allowCredentials",
+  "allowHeaders",
+  "allowMethods",
+  "allowOrigin",
+  "allowOrigins",
+  "exposeHeaders",
+  "maxAge"
+})
+@JsonDeserialize(using = com.fasterxml.jackson.databind.JsonDeserializer.None.class)
+@ToString
+@EqualsAndHashCode
+@Buildable(editableEnabled = false, validationEnabled = false, generateBuilderPackage = false, builderPackage = "io.fabric8.kubernetes.api.builder", refs = {
+  @BuildableReference(ObjectMeta.class),
+  @BuildableReference(ObjectReference.class),
+  @BuildableReference(LabelSelector.class),
+  @BuildableReference(Container.class),
+  @BuildableReference(EnvVar.class),
+  @BuildableReference(ContainerPort.class),
+  @BuildableReference(Volume.class),
+  @BuildableReference(VolumeMount.class)
+})
+public class CorsPolicy implements KubernetesResource {
```

Comment:
Out of curiosity, what's the deal with this class (in every model version).
Why is it manually created?
Is it not generated automatically? (If it is, why isn't there a delete rule in the pom.xml?)

Reply:
In summary, it's because of this sundrio issue: https://github.com/sundrio/sundrio/issues/170
I commented it out in the generator program: https://github.com/fabric8io/kubernetes-client/pull/3613/files#diff-bc93f30af530888e6449c6e0f76d911c332098413e16b51074b62d191560e7edR58

I didn't want to include the comment in the java class because users would see it.

Reply:
Cool.
I think in the main model we don't have this `providedTypes` field, that's why it's confusing. The approach there is to delete the generated type and persist a manually tweaked one.
We should definitely merge this approach there.